### ISSUE TITLE
Mapping with Map( ... ).Index(n) ?

### ISSUE THREAD

**Issue by ftrMorri** [JoshClose/CsvHelper#1124]
I'm getting the following error when using an index based classmap:

> Header matching ['SomeValue'] names at index 0 was not found. If you are expecting some headers to be missing and want to ignore this validation, set the configuration HeaderValidated to null. You can also change the functionality to do something else, like logging the issue.

Here's my code:

```
static void Main(string[] args)
{
    using(TextReader sr = new StringReader("Col1;Col2\r\n1;2"))
    {
        var csv = new CsvReader(sr);
        csv.Configuration.RegisterClassMap<SchemaMap>();
        csv.Configuration.Delimiter = ";";

        var results = csv.GetRecords<Schema>();
        Console.WriteLine(results.Count());
    }
}

public class Schema
{
    public int SomeValue { get; set; }
    public int OtherValue { get; set; }
}

public class SchemaMap : ClassMap<Schema>
{
    public SchemaMap()
    {
        Map(m => m.SomeValue).Index(0);
        Map(m => m.OtherValue).Index(1);
    }
}
```

It looks ok to me, but it looks like CsvHelper is not using index-based mapping, instead it's trying to use "property name - column name" mapping?

**Comment by JoshClose** [comment]
Looks like I don't have a test where `Configuration.HasHeaderRecord = true` and index base mapping are used together.

**Comment by JoshClose** [comment]
This has been fixed in 9.0.1 and is on nuget.
